+bids/layout.m

```diff
@@ -452,114 +382,49 @@ function tolerant_message(use_schema, msg)
 
-        % TODO: this should probably check that the informed_by field does not
-        % already exist in the structure
-        % TODO: This assumes that a file will only be informed by a single file from
-        % this moodality
-        informed_by.(info_src.modality) = file_list{iFile};
-        BIDS.subjects(info_tgt.sub_idx).(info_tgt.modality)(info_tgt.file_idx).informed_by = ...
-            informed_by;
-      end
+    info_src = bids.internal.return_file_info(BIDS, file_list{iFile});
+    file = BIDS.subjects(info_src.sub_idx).(info_src.modality)(info_src.file_idx);
+    metadata = bids.internal.get_metadata(file.metafile);
 
+    % If the file A is intended for file B
+    %   then we update the dependencies.explicit field structrure of file B
+    %   so it contains the fullpath to file A
+    %
+    % This way when one queries info about B, then it is easy to know what
+    % other is present to help with analysis.
+    intended = {};
+    if isfield(metadata, 'IntendedFor')
+      intended = cellstr(metadata.IntendedFor);
     end
 
-  end
-
-end
-
-function [metadata, intended_for] = check_intended_for_exist(subject, filename)
-
-  metadata =  bids.internal.get_metadata(filename);
-
-  intended_for = {};
-
-  if isempty(metadata) || ~isfield(metadata, 'IntendedFor')
-    warning('Missing field IntendedFor for %s', filename);
-    return
-
-  else
-
-    path_intended_for = {};
-
-    if ischar(metadata.IntendedFor)
-      path_intended_for{1, 1} = metadata.IntendedFor;
-
-    elseif isstruct(metadata.IntendedFor)
-      for iPath = 1:length(metadata.IntendedFor)
-        path_intended_for{iPath, 1} = metadata.IntendedFor(iPath); %#ok<*AGROW>
+    for iIntended = 1:numel(intended)
+      dest = fullfile(BIDS.dir, BIDS.subjects(info_src.sub_idx).name, ...
+                      intended{iIntended});
+      if ~exist(dest, 'file')
+        warning(['IntendedFor file ' dest ' from ' file.filename ' not found']);
+        continue
       end
-
+      info_dest = bids.internal.return_file_info(BIDS, dest);
+      BIDS.subjects(info_dest.sub_idx).(info_dest.modality)(info_dest.file_idx) ...
+          .dependencies.explicit{end + 1, 1} = file_list{iFile};
     end
-  end
-
-  % get "subject path" without the session folder (if it exists)
-  subject_path = subject.path;
-  tmp = bids.internal.file_utils(subject_path, 'filename');
-  if strcmp(tmp(1:3), 'ses')
-    subject_path = bids.internal.file_utils(subject_path, 'path');
-  end
-
-  for iPath = 1:size(path_intended_for, 1)
 
-    % create a fullname path for current operating system
-    fullpath_filename = fullfile(subject_path, ...
-                                 strrep(path_intended_for{iPath}, '/', filesep));
-
-    % check if this file is missing
-    if ~exist(fullpath_filename, 'file')
-      warning(['Missing: ' fullpath_filename]);
-
-    else
-      intended_for{end + 1, 1} = fullpath_filename;
-
-    end
   end
 
 end
 
 function perf = manage_asllabeling(perf, pth)
```

Comment:
This `jpg` image also appears in `group` dependency

Reply:
correct

I removed the special way of handling this